src/mlpack/tests/main_tests/range_search_test.cpp

```diff
@@ -0,0 +1,292 @@
+/**
+ * @file range_search_test.cpp
+ * @author Niteya Shah
+ *
+ * Test mlpackMain() of range_search_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "RangeSearchMain";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/range_search/range_search_main.cpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct RangeSearchTestFixture
+{
+ public:
+
+  RangeSearchTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+  ~RangeSearchTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(RangeSearchMainTest, RangeSearchTestFixture);
+/*
+* Check that the correct output is returned for a small synthetic input
+* case
+*/
+BOOST_AUTO_TEST_CASE(SyntheticRangeSearch)
+{
+  cout<<"Synthetic Test 1"<<endl;
+  //Matrix Input is expected in this format
+  arma::mat x={{0, 3, 3, 4, 3, 1},{4, 4, 4, 5, 5, 2},{0, 1, 2, 2, 3, 3}};
+  std::string distance_file="distances.csv";
+  std::string neighbors_file="neighbors.csv";
+  double min_v=0, max_v=3;
+  vector<std::vector<size_t>> neighbor_val={{},{2, 3, 4},{1, 3, 4, 5},{1, 2, 4},
+                                            {1, 2, 3},{2}};
+  vector<std::vector<double>> distance_val={{},{1, 1.73205, 2.23607},
+                                           {1, 1.41421, 1.41421, 3},
+                                           {1.73205, 1.41421, 1.41421},
+                                           {2.23607, 1.41421, 1.41421},
+                                           {3}};
+  SetInputParam("reference", std::move(x));
+  //To prevent warning for lack of definition
+  SetInputParam("min", min_v);
+  SetInputParam("max", max_v);
+  SetInputParam("distances_file", distance_file);
+  SetInputParam("neighbors_file", neighbors_file);
+  cout<<" 1.Parameters Set"<<endl;
+
+  mlpackMain();
+
+  cout<<" 2.Model Created"<<endl;
+  math::Range r(min_v, max_v);
+  vector<vector<size_t>> neighbors;
+  vector<vector<double>> distances;
+  CLI::GetParam<RSModel*>("output_model")->Search(r, neighbors, distances);
+  cout<<" 3.Search Executed"<<endl;
+
+  CheckMatrices(neighbors, neighbor_val, 1e-5);
+  CheckMatrices(distances, distance_val);
+  cout<<" 4.Results Verified"<<endl;
+
+  cout<<"Passed Synthetic Test 1"<<endl<<endl;
+}
+
+BOOST_AUTO_TEST_CASE(ParameterTesting)
```

Comment:
Can you tell me a bit more about what you are testing here?  I am not sure of the difference between what this test is testing and what the previous test is testing.

Reply:
The Search function has two modes , one with query and without . The first ensures that we get the output without the query while the other ensures that we get the correct one with query passed . I felt like I should keep them different but I dont mind merging them into one . Sorry for the confusing name Ill change it .

Reply:
I see, and I agree, we should keep the tests different, but a more descriptive name and documentation for what the test does would help. :)